pkg/maps/metricsmap/doc.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2018Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package metricsmap represents the BPF metrics map in the BPF programs. It is
+// implemented as a hash table containing an entry of different drop reasons and
+// directions. The drop reason 0 is for forwarded packets while a non-zero value
```

Comment:
I don't think it's appropriate to document the specific bit value details in this high-level description of the package. That should instead be above the declaration of the `Key`/`Value` structures.

Consider that this description will appear at the top of the page here:
https://godoc.org/github.com/cilium/cilium/pkg/maps/metricsmap

Compare with this:
https://godoc.org/github.com/cilium/cilium/pkg/maps/lxcmap

Reply:
Sure. I guess I frankly did not see a lot of point in writing this sentence
```
The hashtable can be accessed through the key Key and points which
points to the value Value.
```
Hence altered it to add the above bit specific details

Reply:
How about this:
```
// Package metricsmap represents the BPF metrics map in the BPF programs. It is
// implemented as a hash table containing an entry of different drop reasons and
// directions.The hashtable can be accessed through the key Key and points which
// points to the value Value.
```
Its on the same lines as `lxcmap`

Reply:
Changing to this:
```

// Package metricsmap represents the BPF metrics map in the BPF programs. It is
// implemented as a hash table containing an entry of different drop and forward
// counts for different drop/forward reasons and directions.
```